### PR TITLE
mesh.Mesh.__eq__()

### PR DESCRIPTION
**Pull request by tjlaboss**
In `openmc/mesh.py`, the method `Mesh.__eq__()` contains the following statement:

`elif self._lower_left != mesh2.lower_left and self._upper_right != mesh2.upper_right:`

This results in a ValueError: "The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()". Using `.all()` for the comparison of these lists fixes the error.

### CODE REVIEW

File: openmc/mesh.py [hunk 1]
```diff
@@ -75,8 +75,8 @@ def __eq__(self, mesh2):
             return False
 
         # Check lower left / upper right
-        elif self._lower_left != mesh2.lower_left and \
-             self._upper_right != mesh2.upper_right:
+        elif (self._lower_left != mesh2.lower_left).all() and \
```
Comment (wbinventor) [inline openmc/mesh.py]: Can you apply this same fix to the comparisons of the `width` and `dimension` attributes above?

Comment (nelsonag) [inline openmc/mesh.py]: What about using the EqualityMixin class?  (See the openmc.data package)

Comment (tjlaboss) [inline openmc/mesh.py]: @wbinventor Done.

Comment (nelsonag) [inline openmc/mesh.py]: What about using the EqualityMixin class?  (See the openmc.data package)

Comment (nelsonag) [inline openmc/mesh.py]: (oops)

Comment (wbinventor) [inline openmc/mesh.py]: @nelsonag my understanding is that EqualityMixin has been mostly used in the data submodule, while the array/list comparisons throughout the rest of the API use this more standard approach. Do you believe we should eventually convert everything over to use EqualityMixin? If so, what would be the benefit?

**Comment by nelsonag** [comment]
I view there is no reason it should be only in OpenMC.data and not
elsewhere. The benefit would simply be avoiding the definition of __eq__
and __ne__ everywhere. At the same time in not sure I see a downside to
using it either.

**Comment by paulromano** [comment]
Completely agree with @nelsonag. I don't think there was ever an intention that the mixin only be used in openmc.data since it is generally applicable.

**Comment by wbinventor** [comment]
@nelsonag I see, I didn't notice `EqualityMixin` take over the API last fall! Very nice.

@tjlaboss can you make the `openmc.Mesh` class subclass from the `openmc.mixin.EqualityMixin` class? Then you should be able to remove the `__eq__` method altogether.

**Comment by tjlaboss** [comment]
@nelsonag , @wbinventor : Done, see Pull Req. https://github.com/mit-crpg/openmc/pull/784

**Pull request closed without merging**